### PR TITLE
Install Rails 7.1 and related gems

### PR DESCRIPTION
**Pull request by stevenbarragan**
## Summary
- Rails upgrade to version 7.1
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -56,10 +37,10 @@ class Application < Rails::Application
     config.autoload_paths += Dir[config.root.join('app', 'models', 'elastic_data').to_s]
     config.autoload_paths += Dir[config.root.join('app', 'models', 'logstash_queries').to_s]
 
-    config.middleware.use RejectInvalidRequestUri
-    config.middleware.use DowncaseRoute
-    config.middleware.use AdjustClientIp
-    config.middleware.use FilteredJSONP
+    # config.middleware.use RejectInvalidRequestUri
```
Comment (stevenbarragan) [inline config/application.rb:40]: We still need all the middlewares.

File: config/environments/production.rb [hunk 2]
```diff
@@ -13,61 +13,65 @@
...
 
-  # Do not fallback to assets pipeline if a precompiled asset is missed.
+  # Do not fall back to assets pipeline if a precompiled asset is missed.
   config.assets.compile = false
 
   # Enable serving of images, stylesheets, and JavaScripts from an asset server.
   config.asset_host = ENV['ASSET_HOST']
 
   # Specifies the header that your server uses for sending files.
-  config.action_dispatch.x_sendfile_header = "X-Sendfile" unless ENV['NO_X_SENDFILE_HEADER'].present?
```
Comment (stevenbarragan) [inline config/environments/production.rb:41]: We still need this setting.

File: config/environments/production.rb [hunk 3]
```diff
@@ -13,61 +13,65 @@
...
+  # config.action_dispatch.x_sendfile_header = "X-Sendfile" # for Apache
+  # config.action_dispatch.x_sendfile_header = "X-Accel-Redirect" # for NGINX
 
   # Store uploaded files on the local file system (see config/storage.yml for options).
-  config.active_storage.service = :amazon
+  config.active_storage.service = :local
```
Comment (stevenbarragan) [inline config/environments/production.rb:40]: The upgrade should not change storage service.

File: config/environments/production.rb [hunk 4]
```diff
@@ -13,61 +13,65 @@
...
+
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
   config.force_ssl = true
-  config.ssl_options[:secure_cookies] = true
 
-  # Include generic and useful information about system operation, but avoid logging too much
-  # information to avoid inadvertent exposure of personally identifiable information (PII).
-  config.log_level = ENV.fetch('RAILS_LOG_LEVEL', 'info')
+  # Log to STDOUT by default
+  config.logger = ActiveSupport::Logger.new(STDOUT)
```
Comment (stevenbarragan) [inline config/environments/production.rb:55]: Do not change the logger. We're using the semantic logger and it's set up already.

File: config/environments/production.rb [hunk 5]
```diff
@@ -13,61 +13,65 @@
...
   # Prepend all log lines with the following tags.
   config.log_tags = [ :request_id ]
 
+  # "info" includes generic and useful information about system operation, but avoids logging too much
+  # information to avoid inadvertent exposure of personally identifiable information (PII). If you
+  # want to log everything, set the level to "debug".
+  config.log_level = ENV.fetch("RAILS_LOG_LEVEL", "info")
+
   # Use a different cache store in production.
-  config.cache_store = :redis_cache_store, { url: ENV['REDIS_CACHE_URL'] }
```
Comment (stevenbarragan) [inline config/environments/production.rb:63]: We need the cache store.

File: config/environments/production.rb [hunk 6]
```diff
@@ -13,61 +13,65 @@
...
+  # config.cache_store = :mem_cache_store
 
   # Use a real queuing backend for Active Job (and separate queues per environment).
-  # config.active_job.queue_adapter     = :resque
+  # config.active_job.queue_adapter = :resque
   # config.active_job.queue_name_prefix = "usasearch_production"
 
   config.action_mailer.perform_caching = false
-  config.action_mailer.delivery_method = :sendmail
```
Comment (stevenbarragan) [inline config/environments/production.rb:70]: We need to keep using sendmail. Why remove it?

File: config/initializers/new_framework_defaults_7_1.rb [hunk 7]
```diff
@@ -0,0 +1,280 @@
+# Be sure to restart your server when you modify this file.
+#
+# This file eases your Rails 7.1 framework defaults upgrade.
```
Comment (stevenbarragan) [inline config/initializers/new_framework_defaults_7_1.rb:3]: Can you follow this document? We might already done most of it.

**Review by stevenbarragan — COMMENTED** [review]
We need to keep using the same services and configuration as before.

**Comment by krbhavith** [comment]
I need to remove config/initializers/new_framework_defaults_7_1.rb file after all the config changes are accepted by the reviewer

**Comment by luisgmetzger** [comment]
@krbhavith  - This is looking good but want to confirm that the test will pass for this as there is several changes to the specs.

**Review by luisgmetzger — APPROVED** [review]

**Pull request merged**